cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java

```diff
@@ -23,6 +36,61 @@ public Intellij(IdeContext context) {
     super(context, "intellij", Set.of(Tag.INTELLIJ));
   }
 
+  @Override
+  protected void runIde(String... args) {
+
+    install(true);
+
+    Step stepRun = this.context.newStep("Running IntelliJ");
+    try {
+      ProcessResult result;
+      if (this.context.getSystemInfo().isMac()) {
+        Path ideaPath = getToolPath().resolve("Contents").resolve("MacOS").resolve(IDEA);
+        result = runIntelliJ(ProcessMode.BACKGROUND,
+            CliArgument.prepend(args, "-na", ideaPath.toString(), "--args", this.context.getWorkspacePath().toString()));
+      } else {
+        result = runIntelliJ(ProcessMode.BACKGROUND, CliArgument.prepend(args, this.context.getWorkspacePath().toString()));
+      }
+      if (result.isSuccessful()) {
+        stepRun.success("Running IntelliJ successfully.");
+      } else {
+        stepRun.isFailure();
+      }
+    } catch (Exception e) {
+      stepRun.error(e, "Failed to run IntelliJ.");
+    } finally {
+      stepRun.end();
+    }
+  }
+
+  /**
+   * Runs IntelliJ application.
+   *
+   * @param processMode - the {@link ProcessMode}.
+   * @param args the individual arguments to pass to IntelliJ.
+   * @return the {@link ProcessResult}.
+   */
+  protected ProcessResult runIntelliJ(ProcessMode processMode, String... args) {
+
+    Path toolPath;
+    if (this.context.getSystemInfo().isWindows()) {
+      toolPath = getToolBinPath().resolve(IDEA64_EXE);
+    } else if (this.context.getSystemInfo().isLinux()) {
+      toolPath = getToolBinPath().resolve(IDEA_BASH_SCRIPT);
+    } else {
```

Comment:
If the binary is OS dependent, the correct way would be to override the `getToolBinary()` method.